### PR TITLE
[2/5] [3] cluster: move connect to main impl

### PR DESCRIPTION
**Pull request by 0xWOF**
## Changes

- cluster: move connect function to ClusterConnection impl
- cluster: follow up usage code of connect function

## Effect

- cluster:
    - no public api is changed
    - code is changed but behavior will not be changed

## Note

- this PR is from https://github.com/redis-rs/redis-rs/pull/690
- this PR is depends on https://github.com/redis-rs/redis-rs/pull/706
- you can check actual file changes on https://github.com/0xWOF/redis-rs/pull/15

### CODE REVIEW

File: redis/src/cluster.rs [hunk 1]
```diff
@@ -227,7 +227,7 @@ impl ClusterConnection {
                 _ => panic!("No reach."),
             };
 
-            if let Ok(mut conn) = connect(
+            if let Ok(mut conn) = Self::connect(
```
Comment (djc) [inline redis/src/cluster.rs:230]: This needs to be part of the previous commit; each commit should compile and pass tests.

**Review by djc — COMMENTED** [review]
This mostly looks good to me.

File: redis/src/cluster.rs (hunk 1 above)
Comment (0xWOF) [reply redis/src/cluster.rs:230]: I squash this commit with previous commit, I'll apply this rule to other PRs. Thanks!

**Comment by djc** [comment]
Please rebase. I recommend changing your PRs so each PR targets the previous one's branch. GitHub will update as the original PR gets merged, and the commit view in this PR is actually useful.

**Comment by 0xWOF** [comment]
I didn't know that. Thank you for notify to me. I'll rebase it as soon as possible. And I'll apply this rule to other PRs.

**Comment by 0xWOF** [comment]
@djc I finish rebase. But I can't satisfy this recommend because I have no permission to create branch on redis-rs/redis-rs repository. And PR of `0xWOF:cluster_client_refactor_2_4` to `0xWOF:cluster_client_refactor_2_3` is not PR of redis-rs/redis-rs repository is PR of 0xWOF/redis-rs repository.
> I recommend changing your PRs so each PR targets the previous one's branch instead of main.


Instead, I'll create PR one by one after PR 5. If you know other better way, please share to me.
Thanks!

**Review by djc — APPROVED** [review]

**Comment by djc** [comment]
Creating them one by one seems reasonable to me, and is actually easier on my GitHub notifications.

**Pull request merged**